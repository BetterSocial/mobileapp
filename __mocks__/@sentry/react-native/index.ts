export const addBreadcrumb = jest.fn();
