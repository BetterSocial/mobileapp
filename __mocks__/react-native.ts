import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN