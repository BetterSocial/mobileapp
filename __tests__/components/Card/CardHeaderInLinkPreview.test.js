import * as React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import CardHeaderLink from '../../../src/components/Card/CardHeaderInLinkPreview';
import TestIdConstant from '../../../src/utils/testId';


describe('CardHeaderLink should run correctly', () => {


    afterEach(cleanup)
    it('should match snapshot', () => {
        const {toJSON, getAllByTestId} = render(<CardHeaderLink domain={'https://detik.com'} image='https://detik.jpg' date={'26/08/2022'} />)
        expect(toJSON).toMatchSnapshot()
        expect(getAllByTestId(TestIdConstant.iconDomainProfilePicture)).toHaveLength(1)
                const {getAllByTestId: getNoImageId} = render(<CardHeaderLink domain={'https://detik.com'} image={null} date={'26/08/2022'} />)
        expect(getNoImageId(TestIdConstant.iconDomainProfilePictureEmptyState)).toHaveLength(1)
    })
})