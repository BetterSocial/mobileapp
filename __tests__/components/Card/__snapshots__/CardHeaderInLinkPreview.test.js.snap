// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeaderLink should run correctly should match snapshot 1`] = `[Function]`;
